test/utils_request.js

```diff
@@ -0,0 +1,310 @@
+        });
+
+        it('decompress gzip', async () => {
+            const options = {
+                url: `http://${HOST}:${port}/gzip`,
+
+            };
+
+            const response = await requestBetter(options);
+            expect(response.body)
+                .to
+                .eql(CONTENT);
+        });
+
+        it('decompress deflate', async () => {
+            const options = {
+                url: `http://${HOST}:${port}/deflate`,
+
+            };
+
+            const response = await requestBetter(options);
+            expect(response.body)
+                .to
+                .eql(CONTENT);
+        });
+
+        it('decompress brotli', async () => {
+            const options = {
+                url: `http://${HOST}:${port}/brotli`,
+
+            };
+
+            const response = await requestBetter(options);
+            expect(response.body).to.eql(CONTENT);
+        });
+
+        it('it throws error for 500+', async () => {
+            const options = {
+                url: `http://${HOST}:${port}/500`,
+
+            };
+            let error;
+            try {
+                await requestBetter(options);
+            } catch (e) {
+                error = e;
+            }
+            expect(error.message).to.exist; // eslint-disable-line
+            expect(error.message.includes(ERROR_BODY)).to.be.eql(true);
+        });
+
+        it('it throws error when the body cannot be parsed and the code is 500', async () => {
+            const options = {
+                url: `http://${HOST}:${port}/500/invalidBody`,
+
+            };
+            let error;
+            try {
+                await requestBetter(options);
+            } catch (e) {
+                error = e;
+            }
+            expect(error.message).to.exist; // eslint-disable-line
+        });
+
+        it('it throws error when the body cannot be parsed', async () => {
+            const options = {
+                url: `http://${HOST}:${port}/invalidBody`,
+
+            };
+            let error;
+            try {
+                await requestBetter(options);
+            } catch (e) {
+                error = e;
+            }
+            expect(error.message).to.exist; // eslint-disable-line
+        });
+
+        it('it returns json when 500 even if content-type is different', async () => {
+            const options = {
+                url: `http://${HOST}:${port}/500/json`,
+
+            };
+            let error;
+            try {
+                await requestBetter(options);
+            } catch (e) {
+                error = e;
+            }
+            expect(error.message).to.exist; // eslint-disable-line
+            expect(error.message.includes(JSON_BODY.message)).to.be.eql(true);
+        });
+    });
+
+    describe('Apify.requestAsBrowser', async () => {
+        it('passes crunchbase.com non browser request blocking', async () => {
```

Comment:
When we use something external it must be something absolutely stable. If CrunchBase deploys some anti scraping protection then it breaks our tests

Reply:
I was thinking about it. I think that in general, it will be good to set up a kind of lab with the most common protection strategies and test our approaches on how to bypass them.

Reply:
That's definitely a good idea but it could be a separate project or group of tests.  As we don't want unit tests to fail when some website changes its protection.